tests/models/test_dagrun.py

```diff
@@ -234,8 +234,9 @@ def test_dagrun_deadlock(self):
         ti_op2.set_state(state=State.NONE, session=session)
 
         dr.update_state()
-        self.assertEqual(dr.state, State.RUNNING)
+        self.assertEqual(dr.state, State.FAILED)
```

Comment:
I'm still not 100% on this change. I'll hope to look at this tomorrow (because otherwise I'll merge https://github.com/apache/airflow/pull/6792/files which is going to cause a number of conflicts for one of us)

Reply:
This _does_ look more right now, I just wonder why it was in RUNNING state before?

Reply:
Ok I found the reason. what happens here is that the later task state changes to be "skipped" (which is a bug in my opinion it should be "upstream failed") anyway I deleted this part where it checks if a task state has changed - this is a mistake,  I will return it now.